dali/python/nvidia/dali/_utils/callbacks.py

```diff
@@ -0,0 +1,117 @@
+# Copyright (c) 2021, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
```

Comment:
The name of the module is inadequate. A callback is a function passed as an argument somewhere to be called later (to "call back" the caller). That's far more general than what we do here - namely, handling values that can be passed as `source` parameter of `ExternalSource`. I'd recommend moving it all to `external_source.py` or renaming this file to better reflect what's going on here (like `_external_source_impl.py`?).

Reply:
First, I don't want the external_source.py to grow infinitely, secondly, I want those utilities here to be accessible in few other places that deal with `source`, but are deeper parts of the implementation, rather than the user facing module. So I think grouping them here is better. I can rename, sure.

Reply:
Renamed to external_source_impl.py. `_utils` is already private, so no need to make it doubly private.